### PR TITLE
Introduce transactional (update & delete) methods

### PR DESCRIPTION
**Pull request by rmckirby**
Introduces the ability to action transactional updates/deletes.

### CODE REVIEW

File: scanamo/src/main/scala/org/scanamo/request/ScanamoRequest.scala [hunk 1]
```diff
@@ -72,3 +73,28 @@ case class RequestCondition(
   attributeNames: Map[String, String],
   dynamoValues: Option[DynamoObject]
 )
+
+case class TransactPutItem(
+  tableName: String,
+  item: DynamoValue,
+  condition: Option[RequestCondition]
```
Comment (rmckirby) [inline scanamo/src/main/scala/org/scanamo/request/ScanamoRequest.scala]: Not yet used, but I've left them here should future work want to implement adding the condition expressions

File: alpakka/src/test/scala/org/scanamo/ScanamoAlpakkaSpec.scala [hunk 2]
```diff
@@ -641,4 +641,97 @@ class ScanamoAlpakkaSpec extends FunSpecLike with BeforeAndAfterAll with Matcher
       }
     }
   }
+
+  it("transact table write (update) items") {
```
Comment (rmckirby) [inline alpakka/src/test/scala/org/scanamo/ScanamoAlpakkaSpec.scala]: There's a fair but of duplication between the different specs here, not sure if we see some value in reducing that. For now I've just followed the existing style which seems ok.

Comment (regiskuckaertz) [reply alpakka/src/test/scala/org/scanamo/ScanamoAlpakkaSpec.scala]: Agreed. There is a way around that using some form of `ZManaged` or similar and feeding tables with a random sample of data, then all these tests will be reduced to "laws". One day 😩

**Comment by rmckirby** [comment]
> ❤️ this is amazing work! Thanks a lot for your first contribution, this makes a very useful feature of DynamoDB safe and easy to work with 💪 For heterogeneous transactions, we will need a new construct that can work across tables and accept any kind of CUD operations.

Thanks @regiskuckaertz, I'm happy to contribute!
Do you have a timeframe on when we can expect a new release with these changes in?

**Pull request merged**